sdk/digitaltwins/Azure.DigitalTwins.Core/src/Queries/QueryBuilders/DigitalTwinsQueryBuilder.cs

```diff
@@ -0,0 +1,275 @@
+            topArg.Append(string.Join(", ", args));
+            _selectClauses.Add(new SelectClause(new string[] { topArg.ToString() }));
+
+            return this;
+        }
+
+        /// <summary>
+        /// Used when applying the <see href="https://docs.microsoft.com/en-us/azure/digital-twins/reference-query-clause-select#select-top">TOP()</see> aggregate from the ADT query language. Same functionality as select
+        /// but inserts TOP() into the query structure as well.
+        /// </summary>
+        /// <param name="count">The argument for TOP(), i.e. the number of results to return.</param>
+        /// <returns> Query that contains a select clause. </returns>
+        public DigitalTwinsQueryBuilder SelectTopAll(int count)
+        {
+            // turn into correct format -- eg. SELECT TOP(3)
+            var topArg = new StringBuilder().Append($"{QueryConstants.Top}({count})");
+            _selectClauses.Add(new SelectClause(new string[] { topArg.ToString() }));
+
+            return this;
+        }
+
+        /// <summary>
+        /// Used when applying the <see href="https://docs.microsoft.com/en-us/azure/digital-twins/reference-query-clause-select#select-count">COUNT()</see> aggregate from the ADT query language.
+        /// </summary>
+        /// <returns> Query that contains a select clause. </returns>
+        public DigitalTwinsQueryBuilder SelectCount()
+        {
+            string countArg = $"{QueryConstants.Count}()";
+
+            _selectClauses.Add(new SelectClause(new string[] { countArg }));
+            return this;
+        }
+
+        /// <summary>
+        /// Used when overriding the query builder with the literal query string.
+        /// </summary>
+        /// <param name="customQuery">Query in string format.</param>
+        /// <returns> Query that contains a select clause. </returns>
+        public DigitalTwinsQueryBuilder SelectCustom(string customQuery)
+        {
+            _selectClauses.Add(new SelectClause(new string[] { customQuery }));
+            return this;
+        }
+
+        /// <summary>
+        /// Used to select properties with the desired alias.
+        /// </summary>
+        /// <param name="field">The proper name for the selectable property in the ADT Query Language.</param>
+        /// <param name="alias">The alias to be assigned to the return contents in the query response.</param>
+        /// <returns> Query that contains an aliased select clause. </returns>
+        public DigitalTwinsQueryBuilder SelectAs(string field, string alias)
+        {
+            _selectAsClauses.Add($"{field} {QueryConstants.As} {alias}");
+            return this;
+        }
+
+        /// <summary>
+        /// Adds the FROM clause and its argument to the query via the Clauses component.
+        /// </summary>
+        /// <param name="collection">An enum different collections that users can query from.</param>
+        /// <returns> ADT query a from clause. </returns>
```

Comment:
Let's look for `ADT` in the public code and replace them with DigitalTwins where it makes sense. This can be done in a follow-up PR